app/views/posts/index.html.erb

```diff
@@ -0,0 +1,31 @@
+<p id="notice"><%= notice %></p>
+
+<h1>Posts</h1>
+
+<table>
+  <thead>
+    <tr>
+      
+      <th>Post</th>
+      <th colspan="3">Poster</th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <% @posts.each do |post| %>
+      <tr>
+      
+        <td><%= post.post %></td>
+        <% if user_signed_in? %>
+        <td>from <%= post.user.name %></td>
+        <%end%>
```

Comment:
- View-related logic should be encapsulated in View helpers `(app/helpers/*_helper.rb)`